modules/git/repo_language_stats_gogit.go

```diff
@@ -45,7 +45,8 @@ func (repo *Repository) GetLanguageStats(commitID string) (map[string]int64, err
 
 	var checker *CheckAttributeReader
 
-	if CheckGitVersionAtLeast("1.7.8") == nil {
+	// this code block is refactored from old code "if (git version ...)", keep the block to limit the variable scope
+	{
```

Comment:
I think it might be reasonable to refactor this into a function:

```go
func (repo *Repository) CheckAttributesFor(commitID string) (checker *CheckAttributeReader, deferrable func())
```

Happy to do that refactor if you want.

Reply:
> Hmm... I remember somewhere you commented about CheckLFSVersion causing problems.
> 
> I wonder if It could just be folded into the Init function?

I have tried but got no success at the moment. Since 1.17 is going to be released, I think the big refactoring could be postponed to next milestone, otherwise there will be a lot of code changed, which may affect the 1.17 milestone.

> I think it might be reasonable to refactor this into a function: CheckAttributesFor

Agree it should be refactored. I was thinking about postponing it to next milestone (reason as above). How do you think? I am fine to either plan.